do.sh

```diff
@@ -95,24 +95,61 @@ e2e_assert_jsdeps() {
   fi
 }
 
-e2e_build_library() {
+e2e_build_closure_lib_() {
+  # $1 - Closure entry point
+  # $2 - Filename
+  # $3 - Additional source dir
+  # $4 - [debug|optimized]
+  ENTRY_POINT=$1
+  FNAME=$2
+  SRC_DIRS=( \
+    src \
+    lib/closure-library/closure/goog \
+    lib/closure-library/third_party/closure/goog \
+    lib/closure-templates-compiler lib/zlib.js/src \
+    lib/typedarray )
+  if [ -d "$3" ]; then
+    SRC_DIRS+=("$3")
+  fi
+  jscompile_e2e="$JSCOMPILE_CMD"
+  for var in "${SRC_DIRS[@]}"
+  do
+    jscompile_e2e+=" --js='$var/**.js' --js='!$var/**_test.js' --js='!$var/**_perf.js'"
+  done
+  jscompile_e2e+=" --js='!lib/closure-library/closure/goog/demos/**.js'"
+  if [ "$4" == "debug" ]; then
+     jscompile_e2e+=" --debug --formatting=PRETTY_PRINT -O WHITESPACE_ONLY"
+  elif [ "$4" == "optimized" ]; then
+     jscompile_e2e+=" -O ADVANCED"
+  fi
+  echo -n "."
+  $jscompile_e2e --closure_entry_point "$ENTRY_POINT" --js_output_file "$FNAME"
+}
+
+e2e_build_jsmodule() {
+  echo "Building JS module $1 into $BUILD_DIR/$1.js..."
   e2e_assert_dependencies
   set -e
   e2e_assert_jsdeps
+  mkdir -p "$BUILD_DIR"
+  if [ "$2" == "debug" ]; then
+    echo "Debug mode enabled"
+  fi
+  e2e_build_closure_lib_ $1 "$BUILD_DIR/$1.js" "" $2;
```

Comment:
Would `e2e_build_closure_lib_ $1 "$BUILD_DIR/$1.js" $2 $3` do? So that the full command will become `./do.sh build_jsmodule module_name [additional_dir [debug]]`. I don't want to tie that to templates only, as (a) templates might not be built yet, and I don't want to build them for unrelated targets and (b) people might want to build it against their own codebase.